doc/kubernetes/container-image-management.md

```diff
@@ -0,0 +1,76 @@
+- After we tag latest, we need to remove the previous stopped local container, otherwise when go back to local, systemd service will not create new local container with new version image and will start the stopped container which running with a previous version image.
+
+For the second problem, we need to remove the old version images in time. How to Clean-Up:
+- When: Tag latest will trigger Clean-Up, once the tag latest successfully finished, we will do Clean-Up
+- Clean-Up will clean up all old version images beside the last latest version image(for fallback usage)
+- For image removal, we have three cases to handle:
+    - Case 1: after device(v1) joined k8s cluster, k8s will deploy v1 container to device, kube v1 container  will dry-run. Due to no local to kube replacement happens, Tag-Latest and Clean-Up will not happen. Then we upgrade container to v2 via k8s, kube v2 container will really run and replace the local one. Then the images on device after Tag-Latest should be like below:
+        ```
+        docker-sonic-telemetry:latest(v2)  image_id: kube_v2
+        docker-sonic-telemetry:v1  image_id: local_v1
+        sonick8scue.azurecr.io/docker-sonic-telemetry:v1  image_id: kube_v1
+        sonick8scue.azurecr.io/docker-sonic-telemetry:v2  image_id: kube_v2
+        ```
+        This case we need to remove kube v1 image, because it never really runs, then we tag local v1 image to kube v1 image for fallback prepared, then remove local v1 tag. Then the images after Clean-Up should be like below:
+        ```
+        docker-sonic-telemetry:latest(v2)  image_id: kube_v2
+        sonick8scue.azurecr.io/docker-sonic-telemetry:v1  image_id: local_v1
+        sonick8scue.azurecr.io/docker-sonic-telemetry:v2  image_id: kube_v2
+        ```
+    - Case 2: proceed from the case-1, we have upgrade all devices in one device group container to kube v2 via k8s, then one device get re-imageed, the device is v3, after the device rejoin to k8s cluster, k8s will deploy kube v2 container, kube v2 container will really run and replace the local v3 one. Then the images on device after Tag-Latest should be like below:
+        ```
+        docker-sonic-telemetry:latest(v2)  image_id: kube_v2
+        docker-sonic-telemetry:v3  image_id: local_v3
+        sonick8scue.azurecr.io/docker-sonic-telemetry:v2  image_id: kube_v2
+        ```
+        This case we only need to remove local v3 image, becase it's not last latest version for the k8s daemonset. Then the images after Clean-Up should be like below:
+        ```
+        docker-sonic-telemetry:latest(v2)  image_id: kube_v2
+        sonick8scue.azurecr.io/docker-sonic-telemetry:v2  image_id: kube_v2
+        ```
+        
+    - Case-3, proceed from the case-1, we upgrade container to v3 via k8s, kube v3 container will really run and replace the last kube v2 container. Then the images on device after Tag-Latest should be like below:
```

Comment:
The principle for image cleanup is : keep the version used by kube, and delete others, you can describe the case from this perspective.  Not because "it is not really run"